lars-reimann/api-editor
# refactor: rename `@Unused` to `@Remove`

Closes #518.

### Summary of Changes

Replace the `@Unused` annotation with the `@Remove` annotation.

### Testing Instructions

1. Start the backend:
```sh
cd api-editor
./gradlew run
```
2. Import the API data `data\api\minimal_test_package.json`
3. Add a `@Remove` annotation to some API element
4. Check the filters `annotation:any` and `annotation:@remove` work
5. Remove the `@Remove` annotation
6. Check the "Delete all annotations ` button still works
7. Import the annotation data `data\annotations\minimal_test_package_annotations.json`
8. Check the "Generate adapters` button still works

---------

1. Generate annotations:
```sh
poetry run parse-package annotations -a tests/data/removes/api_data.json -u tests/data/removes/usage_data.json -o out/annotations.json
```
2. Check that a `removes` property exists in the generated JSON file.
